### PR TITLE
mir.ndslice: improve codestyle

### PR DESCRIPTION
**Pull request by wilzbach**
I still had these small improvement laying around - with them I can enable `dscanner` for `mir.ndslice` as it now passes our code style.

It's mostly just > 120 lines, missing `const`, better readability for numbers and undocumented methods.

### CODE REVIEW

File: source/mir/ndslice/selection.d [hunk 1]
```diff
@@ -1713,9 +1713,12 @@ template IotaSlice(size_t N)
 
 // undocumented
 // zero cost variant of `std.range.iota`
+///
 struct IotaMap()
```
Comment (wilzbach) [inline source/mir/ndslice/selection.d:1718]: I don't remember exactly, but you wanted to document it instead of setting it to `private` or `protected`?

Comment (9il) [inline source/mir/ndslice/selection.d:1718]: just undocumented

Comment (wilzbach) [inline source/mir/ndslice/selection.d:1718]: Dscanner rightfully complains about that as you shouldn't have anything that is public and undocumented...

Comment (9il) [inline source/mir/ndslice/selection.d:1718]: Turn it off)

**Comment by wilzbach** [comment]
> Turn it off)

Imho it is one of the more useful switches as I found > 10 occurrences in the codebase of undocumented public methods. So I would like to keep that flag.
Can't we set it to `protected` or `private` as long as you don't want to expose `IotaMap` to the user?

**Comment by 9il** [comment]
> Can't we set it to protected or private as long as you don't want to expose IotaMap to the user?

OK, document it)

**Comment by wilzbach** [comment]
> OK, document it)

like this or with more text?

**Comment by 9il** [comment]
> like this or with more text?

few words

File: source/mir/ndslice/selection.d [hunk 2]
```diff
@@ -1711,11 +1711,15 @@ template IotaSlice(size_t N)
     alias IotaSlice = Slice!(N, IotaMap!());
 }
 
-// undocumented
-// zero cost variant of `std.range.iota`
+/++
+Zero cost variant of `std.range.iota`
```
Comment (9il) [inline source/mir/ndslice/selection.d]: Zero cost variant of `std.range.iota` - this should be with `//` . Just add that this range is used for iotaSlice.

Comment (9il) [inline source/mir/ndslice/selection.d]: Oh, left it, but with small change: 

```
Zero cost variant of `std.range.iota` for iotaSlice.
```

And make `std.range.iota` a link to iota

**Comment by wilzbach** [comment]
updated - i used the new `REF_ALTTEXT` macro , which doesn't work yet with `bootDoc`, but works for Phobos.

**Comment by 9il** [comment]
> REF_ALTTEXT

What is this and would it work for Phobos?

**Comment by wilzbach** [comment]
> What is this and would it work for Phobos?

Yep I recently found it https://github.com/dlang/phobos/pull/4079 - i think it was part of a move to deprecate a lot of old macros within Phobos, but this PR stalled somehow.
It allows to have an alternative text for references - do you know a better macro?

**Comment by 9il** [comment]
> It allows to have an alternative text for references - do you know a better macro?
>  Hide all checks

no

**Pull request merged**